server/CMakeLists.txt

```diff
@@ -21,3 +22,5 @@ target_sources(r-type_server
         ${CMAKE_CURRENT_SOURCE_DIR}/GameLogic.cpp
         ${CMAKE_CURRENT_SOURCE_DIR}/utils/BinaryVector.cpp
 )
+
+target_link_libraries(r-type_server PRIVATE rtype::network)
```

Comment:
This should be under the other target_link_libraries and not under the source files